jetty-websocket/websocket-client/src/test/java/org/eclipse/jetty/websocket/client/ClientCloseTest.java

```diff
@@ -351,6 +352,8 @@ public void testHalfClose() throws Exception
             // client close event on ws-endpoint
             clientSocket.assertReceivedCloseEvent(timeout, is(StatusCode.NORMAL), containsString("From Server"));
         }
+
+        assertThat("Client Open Sessions", client.getOpenSessions(), empty());
```

Comment:
Double check that the `assert` happening in the test thread does not race with some other thread performing the cleanup of the sessions from the `WebSocketClient` - otherwise the test will randomly fail.